addons/survey/tests/test_survey_invite.py

```diff
@@ -231,3 +231,26 @@ def test_survey_invite_token_by_email_nosignup(self):
         answers = self.env['survey.user_input'].search([('survey_id', '=', self.survey.id)])
         self.assertEqual(len(answers), 1)
         self.assertEqual(answers.partner_id.display_name, first_partner.display_name)
+
+    @users('survey_manager')
+    def test_survey_invite_email_from(self):
+        test_survey = self.env['survey.survey'].create({
+            'title': 'Test This Survey'
+        })
+
+        self.env['survey.question'].create({
+            'survey_id': test_survey.id,
+            'title': 'Where is India?',
+            'sequence': 1,
+            'question_type': 'text_box',
+        })
+
+        action = self.survey.action_send_survey()
+        invite_form = Form(self.env[action['res_model']].with_context(action['context']))
+        invite_form.partner_ids.add(self.user_portal.partner_id)
+        invite_form.template_id.write({'email_from':'test name'})
+        invite = invite_form.save()
+        invite.action_invite()
+
+        sender_name = self.env['mail.mail'].sudo().search([], order='id DESC', limit=1)
```

Comment:
We could probably inherit from ``MailCommon`` class (to add in ``TestSurveyInvite`` inherit classes), so that we can mock the mailgateway and use available tools (at least the mock that checks created mail_mail under the mock). It is a better way of doing it :)